specs/online_order_spec.rb

```diff
@@ -1,90 +1,129 @@
 # we effectively get all that testing for free! Here we'll
 # only test things that are different.
 
-xdescribe "OnlineOrder" do
+describe "OnlineOrder" do
   describe "#initialize" do
     it "Is a kind of Order" do
       # Check that an OnlineOrder is in fact a kind of Order
+      #Instatiate your OnlineOrder here
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      online_order = Grocery::OnlineOrder.new(1,products,3)
 
-      # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+      online_order.must_be_kind_of Grocery::Order
     end
 
-    it "Can access Customer object" do
-      # TODO: Your test code here!
-    end
+    # it "Can access Customer object" do
+    #   # TODO: Your test code here!
+    # end
 
     it "Can access the online order status" do
-      # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      online_order = Grocery::OnlineOrder.new(1,products,3)
+
+      online_order.order_status.must_equal :pending
     end
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
-      # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      online_order = Grocery::OnlineOrder.new(1,products,3)
+
+      online_order.total.must_equal 15.36
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      products = {}
+      online_order = Grocery::OnlineOrder.new(2,products,4)
+
+      online_order.total.must_equal 0
     end
   end
 
   describe "#add_product" do
     it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+      products1 = { "banana" => 1.99, "cracker" => 3.00 }
+      products2 = { "Slivered Almonds"=> 22.88, "Wholewheat flour"=> 1.93 }
+      products3 = { "Allspice"=> 64.74, "Bran"=> 14.72 , "UnbleachedFlour"=> 80.59 }
+      online_order1 = Grocery::OnlineOrder.new(1,products1,1, :processing)
+      online_order2 = Grocery::OnlineOrder.new(2,products2,2, :shipped)
+      online_order3 = Grocery::OnlineOrder.new(3,products3,3, :completed)
+
+      online_order1.add_product("salad",4.99).must_be_nil
+      online_order2.add_product("turkey",4.99).must_be_nil
+      online_order3.add_product("gravy",4.99).must_be_nil
     end
 
-    it "Permits action for pending and paid satuses" do
-      # TODO: Your test code here!
+    it "Permits action for pending and paid statuses" do
+      products1 = { "banana" => 1.99, "cracker" => 3.00 }
+      products2 = { "Slivered Almonds"=> 22.88, "Wholewheat flour"=> 1.93 }
+      online_order1 = Grocery::OnlineOrder.new(1,products1,1, :pending)
+      online_order2 = Grocery::OnlineOrder.new(2,products2,2, :paid)
+
+      online_order1.add_product("turkey",7.50).must_be_kind_of TrueClass
+      online_order2.add_product("Slivered Almonds", 22.88).must_be_kind_of FalseClass
```

Comment:
But this test is specifically for checking that you **can** add products to a paid order.  So... good test, wrong place for it.  Instead you should make sure that if you add a product not in `@products` for a paid OnlineOrder it should return true, and `@products` should contain the product. 

You also need to check both orders to verify that they contain the new products.